core/src/HIP/Kokkos_HIP_Space.cpp

```diff
@@ -529,19 +535,17 @@ void SharedAllocationRecord<Kokkos::Experimental::HIPSpace, void>::
     } while (r != &s_root_record);
   }
 #else
+  (void)s;
+  (void)space;
+  (void)detail;
   throw_runtime_exception(
       "Kokkos::Impl::SharedAllocationRecord<HIPSpace>::print_records"
       " only works with KOKKOS_DEBUG enabled");
 #endif
 }
 
-}  // namespace Impl
-}  // namespace Kokkos
-
 /*--------------------------------------------------------------------------*/
 /*--------------------------------------------------------------------------*/
-namespace Kokkos {
-namespace {
```

Comment:
Why did you remove the anonymous namespace?  What warning was issued for it?

Reply:
That this function was never used (because it is in an anonymous namespace and not used in that file). For `CUDA` it also lives in the `Kokkos::Impl` namespace and is not used in the same file.